packages/world-modules/src/modules/erc20-puppet/ERC20Module.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.21;
+
+import { ResourceIds } from "@latticexyz/store/src/codegen/tables/ResourceIds.sol";
+import { ResourceId } from "@latticexyz/store/src/ResourceId.sol";
+import { Module } from "@latticexyz/world/src/Module.sol";
+import { WorldResourceIdLib } from "@latticexyz/world/src/WorldResourceId.sol";
+import { IBaseWorld } from "@latticexyz/world/src/codegen/interfaces/IBaseWorld.sol";
+import { InstalledModules } from "@latticexyz/world/src/codegen/tables/InstalledModules.sol";
+
+import { Puppet } from "../puppet/Puppet.sol";
+import { createPuppet } from "../puppet/createPuppet.sol";
+import { MODULE_NAME as PUPPET_MODULE_NAME } from "../puppet/constants.sol";
+
+import { MODULE_NAME, MODULE_NAMESPACE, MODULE_NAMESPACE_ID, ERC20_REGISTRY_TABLE_ID } from "./constants.sol";
+import { _allowancesTableId, _balancesTableId, _metadataTableId, _erc20SystemId } from "./utils.sol";
+import { ERC20System } from "./ERC20System.sol";
+
+import { ERC20Registry } from "./tables/ERC20Registry.sol";
+import { Balances } from "./tables/Balances.sol";
+import { Allowances } from "./tables/Allowances.sol";
+import { Metadata, MetadataData } from "./tables/Metadata.sol";
+
+contract ERC20Module is Module {
+  error ERC20Module_InvalidNamespace(bytes14 namespace);
+
+  function getName() public pure override returns (bytes16) {
+    return MODULE_NAME;
+  }
+
+  /**
+   * Register systems and tables for a new ERC20 token in a given namespace
+   */
+  function _registerERC20(bytes14 namespace) internal {
+    // Register the tables
+    Allowances.register(_allowancesTableId(namespace));
+    Balances.register(_balancesTableId(namespace));
+    Metadata.register(_metadataTableId(namespace));
+
+    // Register a new ERC20System
+    IBaseWorld(_world()).registerSystem(_erc20SystemId(namespace), new ERC20System(), true);
+  }
+
+  function _requireDependencies() internal view {
+    // If the PuppetModule is not installed yet, install it
+    if (InstalledModules.get(PUPPET_MODULE_NAME, keccak256(new bytes(0))) == address(0)) {
```

Comment:
since we're starting to use this more, might be nice to lift this into `Module` as a helper function
e.g. `isInstalled()` or `isInstalled(module)`

Reply:
https://github.com/latticexyz/mud/issues/1852